utils/policy_roles_adapter.js

```diff
@@ -67,25 +67,32 @@ const MAPPING = {
     restrictPostDelete: {
         all: [
             {roleName: 'channel_user', permission: Permissions.DELETE_POST, shouldHave: true},
-            {roleName: 'channel_admin', permission: Permissions.DELETE_POST, shouldHave: false},
-            {roleName: 'channel_admin', permission: Permissions.DELETE_OTHERS_POSTS, shouldHave: false},
             {roleName: 'team_admin', permission: Permissions.DELETE_POST, shouldHave: true},
             {roleName: 'team_admin', permission: Permissions.DELETE_OTHERS_POSTS, shouldHave: true}
         ],
         team_admin: [
             {roleName: 'channel_user', permission: Permissions.DELETE_POST, shouldHave: false},
-            {roleName: 'channel_admin', permission: Permissions.DELETE_POST, shouldHave: false},
-            {roleName: 'channel_admin', permission: Permissions.DELETE_OTHERS_POSTS, shouldHave: false},
             {roleName: 'team_admin', permission: Permissions.DELETE_POST, shouldHave: true},
             {roleName: 'team_admin', permission: Permissions.DELETE_OTHERS_POSTS, shouldHave: true}
         ],
         system_admin: [
             {roleName: 'channel_user', permission: Permissions.DELETE_POST, shouldHave: false},
-            {roleName: 'channel_admin', permission: Permissions.DELETE_POST, shouldHave: false},
-            {roleName: 'channel_admin', permission: Permissions.DELETE_OTHERS_POSTS, shouldHave: false},
             {roleName: 'team_admin', permission: Permissions.DELETE_POST, shouldHave: false},
             {roleName: 'team_admin', permission: Permissions.DELETE_OTHERS_POSTS, shouldHave: false}
         ]
+    },
+
+    enableOnlyAdminIntegrations: {
+        true: [
+            {roleName: 'team_user', permission: Permissions.MANAGE_WEBHOOKS, shouldHave: false},
+            {roleName: 'team_user', permission: Permissions.MANAGE_SLASH_COMMANDS, shouldHave: false},
+            {roleName: 'system_user', permission: Permissions.MANAGE_OAUTH, shouldHave: false}
+        ],
+        false: [
+            {roleName: 'team_user', permission: Permissions.MANAGE_WEBHOOKS, shouldHave: true},
+            {roleName: 'team_user', permission: Permissions.MANAGE_SLASH_COMMANDS, shouldHave: true},
+            {roleName: 'system_user', permission: Permissions.MANAGE_OAUTH, shouldHave: true}
+        ]
```

Comment:
@grundleborg We may want to review these new mappings together. I took a stab at what I think it might be and also assumed that the `MANAGE_OTHERS_WEBHOOKS`, `MANAGE_SYSTEM_WIDE_OAUTH`, and `MANAGE_OTHERS_SLASH_COMMANDS` were not updated.

Reply:
Yes, I've checked this in several ways now, and I think it's correct!